File: Python/02_strings.py
#!/usr/bin/env python3

def strings():
    """08_strings.py - Strings examples"""
    print("=== STRINGS ===")
    # Basic string creation and concatenation
    name = "Alice"
    greeting = "Hello, " + name + "!"
    print(greeting)

    # Modern f-strings (recommended)
    age = 30
    print(f"My name is {name} and I am {age} years old.")

    # Common string methods
    text = "   Python Programming   "
    print("Uppercase:          ", text.upper())
    print("Lowercase:          ", text.lower())
    print("Stripped:           ", text.strip())
    print("Split into words:   ", text.strip().split())
    print("Replace:            ", text.replace("Python", "Rust"))
    print("Starts with 'Py':   ", text.strip().startswith("Python"))

    # String slicing
    lang = "Python"
    print("First 3 chars:      ", lang[:3])
    print("Last 3 chars:       ", lang[-3:])
    print("Reversed:           ", lang[::-1])
    print("Every other char:   ", lang[::2])

def main():
    strings()

if __name__ == '__main__':
    main()

File: JavaScript/02_strings.js
#!/usr/bin/env node

function strings() {
    console.log("=== STRINGS ===");
    const name = "Alice";
    console.log("Hello, " + name + "!");

    const age = 30;
    console.log(`My name is ${name} and I am ${age} years old.`);

    let text = "   JavaScript Programming   ";
    console.log("Uppercase:", text.toUpperCase());
    console.log("Lowercase:", text.toLowerCase());
    console.log("Stripped:", text.trim());
    console.log("Split:", text.trim().split(" "));
    console.log("Replace:", text.replace("JavaScript", "TypeScript"));

    const lang = "JavaScript";
    console.log("First 4 chars:", lang.slice(0, 4));
    console.log("Reversed:", lang.split("").reverse().join(""));
}

function main() {
    strings();
}

main();
